blacklight-cornell/app/models/search_builder.rb

```diff
@@ -33,62 +33,64 @@ def sortby_callnum user_parameters
+    solr_parameters[:fl] = "*" if blacklight_params["controller"] == "bookmarks" || blacklight_params["format"].present? || blacklight_params["controller"] == "book_bags"
+
+    # Standard blacklight_params from advanced search form:
+    # {
+    #   "advanced_query"=>"yes", 
+    #   "boolean_row"=>{"1"=>"AND"},
+    #   "commit"=>"Search", 
+    #   "op_row"=>["AND", "AND"], 
+    #   "q_row"=>["test", ""], 
+    #   "range_end"=>"2025", 
+    #   "range_field"=>"pub_date_facet", 
+    #   "range_start"=>"0", 
+    #   "search_field"=>"advanced", 
+    #   "search_field_row"=>["all_fields", "all_fields"], 
+    #   "sort"=>"score desc, pub_date_sort desc, title_sort asc", 
+    #   "utf8"=>"✓", 
+    #   "controller"=>"catalog", 
+    #   "action"=>"range_limit"
+    # }
+    # For non-standard advanced search params, use presence of q_row to determine if query should be processed as advanced search
     if blacklight_params[:q_row].present?
+      # Clean up/reset non-standard advanced search params
       blacklight_params.delete('q')
+      solr_parameters[:search_field] = 'advanced'
 
-      my_params = make_adv_query(blacklight_params)
-      spellstring = ""
-      if !my_params[:q_row].nil?
-        blacklight_params[:q_row].each do |term|
-          spellstring += term += ' '
-          #spellstring  += term +  ' '
-        end
-      else
-      end
-      user_parameters[:q] = my_params[:q] #blacklight_params[:q]
- #     blacklight_params[:q] = user_parameters[:q]
-      user_parameters[:search_field] = "advanced"
-      #user_parameters["mm"] = "100"
-      user_parameters["mm"] = "1"
-      user_parameters["defType"] = "edismax"
+      solr_parameters[:q] = build_advanced_search_query(blacklight_params)
     else # simple search code below
       if blacklight_params[:q].nil?
         blacklight_params[:q] = ''
         if !blacklight_params[:f].nil?
-          	user_parameters[:fq] = []
-          	fq_string = ""
-          	blacklight_params[:f].each do |key, value|
-
-          	  	value.each do |val|
-                   if (val == 'last_1_week' or val == 'last_1_month' or val == 'last_1_years')
-						if val == 'last_1_week'
-          	    			fq_string = 'acquired_dt:[NOW-14DAY TO NOW-7DAY ]'
-                    	else 
-                      		if val == 'last_1_month'
-                      			fq_string = 'acquired_dt:[NOW-30DAY TO NOW-7DAY ]'
-                      		else
-                      			if value[0] == 'last_1_years'
-                      				fq_string = 'acquired_dt:[NOW-1YEAR TO NOW-7DAY]'
-                        		end
-                      		end
-                    	end                   
-                    else
-				      fq_string = '{!term f=' + key + '}' + val
-				    end
-				  	user_parameters[:fq] << fq_string
-          	    	blacklight_params[:fq] = user_parameters[:fq]
-          	  	end
-          	 # end
-          	end
+          solr_parameters[:fq] = []
```

Comment:
Note: I did not actually make any changes from here to the end of `#advsearch` except for fixing the padding and renaming `user_parameters` to `solr_parameters` to make it more obvious what this data is. (Just an FYI to make reviewing this PR easier.)